api/src/main/java/io/opentelemetry/common/Attributes.java

```diff
@@ -155,12 +155,32 @@ public Attributes build() {
       return sortAndFilterToAttributes(data.toArray());
     }
 
+    private boolean doNotAdd(String key, AttributeValue value) {
+      if (key == null || key.length() == 0) {
+        return true;
+      }
+      if (value == null || value.isEmpty()) {
+        int index = data.indexOf(key);
+        if (index == -1) {
+          return true;
+        }
+        // Remove key/value pair
+        data.remove(index);
+        data.remove(index);
```

Comment:
Have this method, which looks like a simple query method, also have the side effect of removing data is very surprising. The name of this method should reflect that it is potentially an update operation. Also, how does this handle the fact that the key might be in the data twice at this point? Don't we need to iterate through the entire array to remove all instances of the key?

Reply:
You are right! I have overlooked that dedup is done as the last thing in the builder.